src/modules/villages/village_repository.ts

```diff
@@ -24,10 +24,42 @@ export namespace Village {
 
     if (requestQuery.name) query.where('villages.name', 'LIKE', `%${requestQuery.name}%`)
     if (requestQuery.level) query.where('villages.level', requestQuery.level)
+    if (isRequestBounds(requestQuery)) query.whereRaw(getWherePolygon(requestQuery))
 
     return query
   }
 
+  const pointRegexRule = (point: string) => {
+    const pointRegex = /^(-?\d+(\.\d+)?), \s*(-?\d+(\.\d+)?)$/
+    return pointRegex.test(point)
+  }
+
+  const getWherePolygon = (requestQuery: Entity.RequestQuery) => {
+    return `ST_CONTAINS(ST_GEOMFROMTEXT('${getPolygon(requestQuery)}'), villages.location)`
+  }
+
+  const isRequestBounds = (requestQuery: Entity.RequestQuery) => {
+    return requestQuery?.bounds?.ne &&
+    requestQuery?.bounds?.sw &&
+    pointRegexRule(requestQuery.bounds.ne) &&
+    pointRegexRule(requestQuery.bounds.sw)
+  }
+
+  const getPolygon = (requestQuery: Entity.RequestQuery) => {
+    const { ne, sw } = requestQuery.bounds
+
+    const boundsNE = ne.trimStart().trimEnd().split(/[, ]+/)
+    const boundsSW = sw.trimStart().trimEnd().split(/[, ]+/)
+
+    return `POLYGON((
+      ${boundsNE[0]} ${boundsNE[1]},
+      ${boundsNE[0]} ${boundsSW[1]},
+      ${boundsSW[0]} ${boundsSW[1]},
+      ${boundsSW[0]} ${boundsNE[1]},
+      ${boundsNE[0]} ${boundsNE[1]}
```

Comment:
Can I ask for the reference, kang @ayocodingit the algorithm is polygon like this?
I'm still new to map algorithm.

Reply:
![image](https://user-images.githubusercontent.com/41193120/140247848-ed77fd1b-1386-4794-8fb2-7af641a414e3.png)


the point is that the algorithm exists because from gmaps only 2 points are given, as shown below

![image](https://user-images.githubusercontent.com/41193120/140247910-753b197a-5c63-48d6-bf93-4a77d09f96cd.png)

Reply:
thanks @ayocodingit 👍